### PR TITLE
fix: osidb-bot robustness improvements

### PR DESCRIPTION
**Pull request by kdudka**
## What
Three fixes to improve osidb-bot robustness: catch unhandled per-CVE exceptions so they don't kill the asyncio pool, relax the CVE ID validation regex, and log the actual number of successfully processed CVEs.

## Why
An unhandled `ValidationError` from a CVE ID with more than 7 digits in the sequence number caused the entire batch to abort. Additionally, the "processed N CVEs" log message always reported the total count rather than the number actually processed.

## How to Test
```bash
make check test
uv run aegis osidb-bot CVE-2026-1244567823 --max-retries=3 --state-file=state.json
```

## Related Tickets
Related: https://redhat.atlassian.net/browse/AEGIS-462

## Summary by Sourcery

Improve osidb-bot robustness when processing CVEs and accurately report processing outcomes.

Bug Fixes:
- Ensure per-CVE exceptions do not abort the entire asyncio batch when processing a list of CVEs.
- Count and log the actual number of successfully processed CVEs instead of always reporting the total input count.

Enhancements:
- Have flaw processing return a boolean success flag that is propagated through CVE processing to support more accurate batch accounting.
- Relax CVE ID validation to accept sequence numbers up to 16 digits.

Tests:
- Update CVE ID validator test to reflect the relaxed CVE ID pattern.

### CODE REVIEW

File: tests/test_data_models.py [hunk 1]
```diff
@@ -19,4 +19,4 @@ def test_cveid_validator():
     with pytest.raises(ValidationError) as excinfo:
         cve_id = "BAD-CVE-4"
         assert cveid_validator.validate_python(cve_id)
-    assert "String should match pattern '^CVE-[0-9]{4}-[0-9]{4,7}$'" in str(excinfo)
+    assert "String should match pattern '^CVE-[0-9]{4}-[0-9]{4,16}$'" in str(excinfo)
```
[the remark this answers is not in the document]
Comment (kdudka) [reply tests/test_data_models.py:22]: This is a corner case really.  We do not have such long CVE IDs in OSIDB.

File: src/aegis_ai/osidb_bot/bot.py [hunk 2]
```diff
@@ -352,12 +353,12 @@ async def do(self) -> None:
...
 
         # apply suggestions
-        all_ok = await self.apply_suggestions()
+        all_ok: bool = await self.apply_suggestions()
 
         if self.read_only:
             msg = f"read-only mode, skipping OSIDB update of {self.updated_fields}"
             self._warn(msg)
-            return
+            return False
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai/osidb_bot/bot.py:361]: That is the correct expected behavior.  If a suggestion failed e.g. because the provided context was insufficient, it is unlikely that it would succeed 5 minutes later.  We only retry processing of CVEs when something unexpected (but recoverable) happens, such as Mid-Air collision while updating OSIDB data.

**Review by superbuggy — APPROVED** [review]

File: docs/openapi.yml [hunk 3]
```diff
@@ -136,6 +136,73 @@ paths:
...
+                agent:
+                  type: string
+                  enum:
+                  - public
+                  - redhat
+                  description: >-
+                    Agent to use for analysis. 'public' removes OSIDB tools so the
+                    LLM analyses user-provided text only. 'redhat' includes OSIDB
+                    tools (default when omitted; governed by AEGIS_WEB_FEATURE_AGENT).
+              additionalProperties: true
```
[the remark this answers is not in the document]
Comment (kdudka) [reply docs/openapi.yml:205]: Let me reflect the updated CVE ID regex.  Everything else is going to be fixed via #759.

[the remark this answers is not in the document]
Comment (kdudka) [reply docs/openapi.yml:205]: Fixed in 945e2a06d14d865261b4ff519050457b8e2e2837.

File: evals/ghsa_cache/CVE-2025-39816.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                    "events": [
+                        {
+                            "introduced": "6.13.0"
+                        },
+                        {
+                            "fixed": "6.12.81"
+                        },
+                        {
+                            "fixed": "6.16.5"
+                        }
```
[the remark this answers is not in the document]
Comment (kdudka) [reply evals/ghsa_cache/CVE-2025-39816.json:108]: This is just external data used as fixtures for the evaluation suite.
